### PR TITLE
Feature/adaptive learning rate

### PR DESCRIPTION
**Pull request by leoTiez**

### CODE REVIEW

File: RCNN_tf.py [hunk 1]
```diff
@@ -373,6 +376,7 @@ def __init__(
                 name='rcl_layer_1'
             )
 
+
```
Comment (bt-s) [inline RCNN_tf.py]: Please remove, sir.

File: RCNN_tf.py [hunk 2]
```diff
@@ -530,10 +545,14 @@ def train(self, train_data_feats, train_data_labels, val_data_feats,
...
                     print '\r {:.1f}%'.format(progress), '\t{0}> '.format('#' * int(progress)),
 
+                    if last_update_of_performance >= performance_update_threshold:
+                        learning_rate *= learning_rate * adaptive_learning_factor
+
                     accuracies, _, cost_ = sess.run(
                         [self.summaries, self.optimiser, self.cross_entropy],
                         feed_dict={
-                            self.rate_placeholder: 0.2
+                            self.rate_placeholder: 0.2,
```
Comment (bt-s) [inline RCNN_tf.py]: Should be an input parameter to ```train()```.

**Review by bt-s — APPROVED** [review]
Molto bene!

File: RCNN_tf.py (hunk 1 above)
Comment (leoTiez) [reply RCNN_tf.py]: Yes, sir!

File: RCNN_tf.py (hunk 2 above)
Comment (leoTiez) [reply RCNN_tf.py]: Indeed, I will change that!

**Review by adrianc-a — APPROVED** [review]
Approved with bas' requested changes.
